test/cql-pytest/test_describe.py

```diff
@@ -0,0 +1,166 @@
+# Copyright 2022-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+#############################################################################
+# Tests for server-side describe
+#############################################################################
+
+import pytest
+from util import new_test_table, new_type, new_function, new_aggregate
+
+# Extract names from listing describe result
+def extract_names(result):
+    return list(map(lambda r: r.name, result))
+
+# Extract descriptions from describe result and join them to one string
+def extract_describe(result):
+    descs = list(map(lambda r: r.create_statement, result))
+    desc = " ".join(descs)
+    return desc
+
+# Test that `DESC KEYSPACES` contains newly created user keyspace
+def test_desc_keyspaces(cql, test_keyspace):
+        result = cql.execute("DESC KEYSPACES")
+        names = extract_names(result)
+        assert test_keyspace in names
+
+# Test that `DESC KEYSPACE {name}` contains create statements for keyspace and table
+# and `DESC ONLY KEYSPACE {name}` contains only create statement for keyspace
+def test_desc_keyspace(cql, test_keyspace):
+    with new_test_table(cql, test_keyspace, "id int primary key") as table:
+        result = cql.execute(f"DESC KEYSPACE {test_keyspace}")
+        desc = extract_describe(result)
+
+        assert f"CREATE KEYSPACE {test_keyspace}" in desc
+        assert f"CREATE TABLE {table}" in desc
+
+        only_result = cql.execute(f"DESC ONLY KEYSPACE {test_keyspace}")
+        only_desc = extract_describe(only_result)
+
+        assert f"CREATE KEYSPACE {test_keyspace}" in only_desc
+        assert f"CREATE TABLE {table}" not in only_desc
+
+# Test that `DESC TABLES` contains some system table
+# and newly created table
+def test_desc_tables(cql, test_keyspace):
+    result = cql.execute("DESC TABLES")
+    names = extract_names(result)
+
+    assert "local" in names
+    assert "sstable_activity" in names
+
+    with new_test_table(cql, test_keyspace, "id int primary key") as table:
+        table_name = table.split(".")[1]
+        cql.execute(f"use {test_keyspace}")
+        ks_result = cql.execute("DESC TABLES")
+        ks_names = extract_names(ks_result)
+
+        assert "local" not in ks_names
+        assert "sstable_activity" not in ks_names
+        assert table_name in ks_names
+
+# Test that `DESC TABLE {name}` contains create statement for table
+def test_desc_table(cql, test_keyspace):
+    with new_test_table(cql, test_keyspace, "id int primary key") as table:
+        table_name = table.split(".")[1]
```

Comment:
This split is unnecessary - you never use table_name separately before - just `{test_keyspace}.{table_name}`. But what's nothing more than "table" :-) So you can just "table" everywhere below.